staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_patch_test.go

```diff
@@ -67,7 +67,66 @@ func TestAdjustClusterNameIfWildcard(t *testing.T) {
 			}
 			expected := "root:org:ws"
 
-			clusterName := adjustClusterNameIfWildcard(cluster, !tc.builtInType, tc.prefix, key)
+			clusterName := adjustClusterNameIfWildcard(genericapirequest.Shard(""), cluster, !tc.builtInType, tc.prefix, key)
+			if e, a := expected, clusterName.String(); e != a {
+				t.Errorf("expected: %q, actual %q", e, a)
+			}
+		})
+	}
+}
+
+func TestAdjustClusterNameIfWildcardWithShardSupport(t *testing.T) {
+	tests := map[string]struct {
+		clusterWildcard bool
+		shardWildcard   bool
+		prefix          string
+		builtInType     bool
```

Comment:
can this be pure table test with inputs and outputs? Irritating there is no "expected*" in here. Just put `genericapirequest.Cluster` and `genericapirequest.Shard` in here, no need for abstraction.

Reply:
updated